fl4health/privacy_mechanisms/dpDiscreteGaussian.py

```diff
@@ -0,0 +1,132 @@
+import math
+from typing import List
+
+import torch
+
+
+def Bernoulli_exp(gamma: float) -> int:
+    """
+    Draws a sample from Bernoulli(exp(-gamma))
+
+    Args:
+        gamma (float): A non-negative parameter of type float.
+
+    Returns:
+        int: A Bernoulli sample, either 0 or 1
+
+    Reference:
+        Algorithm 2 of "The Discrete Gaussian for Differential Privacy"
+        https://proceedings.neurips.cc/paper/2020/file/b53b3a3d6ab90ce0268229151c9bde11-Paper.pdf
+    """
+
+    assert gamma >= 0  # ensures probabiity is not over 1
+
+    if 0 <= gamma <= 1:
+        K, g = 1, torch.tensor(float(gamma))
+        sample_A = torch.bernoulli(g)
+
+        while sample_A == 1:
+            K += 1
+            sample_A = torch.bernoulli(g / K)
+
+        return K % 2
+
+    # if gamma > 1
+    G, p = math.floor(gamma), torch.exp(torch.tensor(-1))
+
+    for _ in range(G):
+        sample_B = torch.bernoulli(p)
```

Comment:
Based on my reading of Algorithm 2 from the reference paper, this is supposed to be a recursive call, similar to sample_C. That is, this line should be `Bernoulli_exp(p.item())` rather than an explicit bernoulli sample.